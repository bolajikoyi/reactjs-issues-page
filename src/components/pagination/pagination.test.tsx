/* eslint-disable testing-library/no-node-access */
import React from 'react';
import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Pagination from './pagination';

afterEach(cleanup);

let data = {
    pageInfo: {
        endCursor: "Y3Vyc29yOnYyOpHOQwZW6Q==",
        hasNextPage: false,
        hasPreviousPage: true,
        startCursor: "Y3Vyc29yOnYyOpHOP9wfNA==",
        __typename: "PageInfo"
    },
    pagination: jest.fn()
}

describe('Pagination Component', ()=>{

    test('Expect component to be created', async () => {
        render(<Pagination action={data}/>)
        expect(Pagination).toBeTruthy()
    });

    test('previous button is in document and enabled', async () => {
        render(<Pagination action={data}/>)
        expect(screen.getByTestId('previous')).toBeInTheDocument();
        expect(screen.getByTestId('previous')).toBeEnabled();
    });

    test('next button is in document and disabled', async () => {
        render(<Pagination action={data}/>)
        expect(screen.getByTestId('next')).toBeInTheDocument();
        expect(screen.getByTestId('next')).toBeDisabled();
    });

    test('matches snapshot', ()=> {
      const tree = renderer.create(<Pagination action={data}/>).toJSON();
      expect(tree).toMatchSnapshot();
  })
})

