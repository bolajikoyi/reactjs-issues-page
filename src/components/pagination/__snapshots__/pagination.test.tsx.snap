// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component matches snapshot 1`] = `
<div
  className="pagination-buttons"
>
  <button
    className="previous"
    data-testid="previous"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="fas fa-less-than"
    />
     Previous
  </button>
  <button
    className="next"
    data-testid="next"
    disabled={true}
    onClick={[Function]}
  >
    Next 
    <i
      className="fas fa-greater-than"
    />
  </button>
</div>
`;
